go-tutorial.md

```diff
@@ -0,0 +1,1233 @@
+---
+nav_order: 3
+---
+
+# Go and Bazel
+
+## This tutorial covers
+
+- Background of bazel and it's Go support
+- Creating a basic Go project for the tutorial
+- Implementing a WORKSPACE and BUILD.bazel files
+- Using gazelle to generate more WORKSPACE and BUILD.bazel updates
+- Utilzing different Bazel commands
+- An overview of gazelle and Go dependency management
+- Understanding the contents of the WORKSPACE and BUILD.bazel files
+- Creating new internal dependencies and using gazelle to update bazel
+- Adding an a new external Go dependency and the Go vendoring
+- Running and implementing Go tests with bazel
+- Covering other rules in rules_go
+
+## About Bazel and Go
+
+This tutorial is going to cover how bazel support the programming language Go.
+
+[Bazel](https://bazel.build) is an open-source build and test application that supports the software development lifecycle. Bazel strives
+to allow a developer to have hermetic and deterministic builds, testing, packaging and deployment.  This build
+tool supports multiple languages and builds binaries for multiple platforms.
+
+Of the multiple languages that bazel supports includes [Go](https://go.dev). Go is an open-source programming languages
+that was created by Google.
+
+One of the concepts within bazel is a rule.  A bazel rule defines a series of actions and outputs. Toolchains is another
```

Comment:
Outside of a code snippet, should `bazel` always be capitalized (e.g., `Bazel`)?

Reply:
Updated both the words "Bazel" and "Gazelle"